src/Streamer.cxx

```diff
@@ -84,225 +80,240 @@ int FileWriter::Streamer::get_topic_partitions(const std::string &topic) {
-                     [&](std::pair<std::string, std::string> &item) -> bool {
-                       return this->set_conf_opt(conf, item);
-                     }),
-      kafka_options.end());
+  kafka_options.erase(std::remove_if(kafka_options.begin(), kafka_options.end(),
+                                     [&](option_t &item) -> bool {
+                                       return this->set_conf_opt(conf, item);
+                                     }),
+                      kafka_options.end());
   if (!kafka_options.empty()) {
     for (auto &item : kafka_options) {
       LOG(3, "Unknown option: {} [{}]", item.first, item.second);
     }
   }
+  return conf;
+}
 
-  if (!(_consumer = RdKafka::KafkaConsumer::create(conf.get(), errstr))) {
-    LOG(0, "Failed to create consumer: {}", errstr);
-    status_.emplace("status.consumer", -1);
-  } else {
-    status_.emplace("status.consumer", 1);
+void FileWriter::Streamer::initialize_streamer(
+    FileWriter::Streamer::Options &filewriter_options) {
+  filewriter_options.erase(std::remove_if(filewriter_options.begin(),
+                                          filewriter_options.end(),
+                                          [&](option_t &item) -> bool {
+                                            return this->set_streamer_opt(item);
+                                          }),
+                           filewriter_options.end());
+  if (!filewriter_options.empty()) {
+    for (auto &item : filewriter_options) {
+      LOG(3, "Unknown option: {} [{}]", item.first, item.second);
+    }
   }
+  return;
+}
 
-  if (!topic_name.empty()) {
-    status_.emplace("status.metadata", 1);
-    if (get_metadata() != 0) {
-      LOG(1, "Unable to retrieve metadata");
-      status_["status.metadata"] = -1;
-    }
-    status_.emplace("status.topic_partition", 1);
-    if (get_topic_partitions(topic_name) < 0) {
-      LOG(1, "Unable to build TopicPartitions structure");
-      status_["status.topic_partition"] = -1;
-    }
-    for (auto &i : _tp) {
-      i->set_offset(_offset.value());
-    }
+FileWriter::Streamer::Streamer(
+    const std::string &broker, const std::string &topic_name,
+    FileWriter::Streamer::Options kafka_options,
+    FileWriter::Streamer::Options filewriter_options) {
+
+  s_.run_status(Error(ErrorCode::not_initialized));
+  if (topic_name.empty() || broker.empty()) {
+    LOG(0, "Broker and topic required");
+    s_.run_status(Error(ErrorCode::not_initialized));
+    return;
+  }
+  kafka_options.push_back(
+      FileWriter::Streamer::option_t{"metadata.broker.list", broker});
+  kafka_options.push_back(
+      FileWriter::Streamer::option_t{"api.version.request", "true"});
+  kafka_options.push_back(
+      FileWriter::Streamer::option_t{"group.id", topic_name});
+
+  connect_ = std::thread([&] {
+    this->connect(topic_name, kafka_options, filewriter_options);
+    return;
+  });
+  std::this_thread::sleep_for(milliseconds(10));
```

Comment:
The `wait` timing out gives whatever launches that thread the information that connecting is taking too long, that could also be advantageous.